documentation/docs/25-build-and-deploy/40-adapter-node.md

```diff
@@ -162,6 +166,45 @@ MY_CUSTOM_ORIGIN=https://my.site \
 node build
 ```
 
+## Socket Activation
+
+Most Linux operating systems today use a modern process manager called systemd to start the server and run and manage services. You can configure your server to allocate a socket and start and scale your app on demand. This is called [socket activation](http://0pointer.de/blog/projects/socket-activated-containers.html). In this case the OS will pass two environment variables to your app — `LISTEN_PID` and `LISTEN_FDS`. The adapter will verify that these variables are correct and then listen on file descriptor 3 which refers to a systemd socket unit that you will have to create.
```

Comment:
Will these environment variables _always_ be called that? The way this is written, I believe the adapted code will expect them to be prefixed with `envPrefix` if you've passed that config to the adapter. Is that desirable? Will that work?

If there's no way to control what env var names the OS expose to the app, we should either add a note not to use `envPrefix` in that case, or we should add different handling for `LISTEN_PID` and `LISTEN_FDS` and add a note to the docs about that. I'm not entirely comfortable with that second solution, though.

Reply:
Oh~! right. They should never prefixed. They will always be called `LISTEN_PID` and `LISTEN_FDS`. I guess just not using prefix in that case is probably the better solution.

Reply:
Hey! Sorry, I'm just getting back to reviewing this now. As far as I can tell, since you're using the `env()` function, `LISTEN_PID` and `LISTEN_FDS` will still be looked for in their prefixed versions.

Reply:
Oh, I see you added a note in the docs about this. I don't know how I feel about the resolution being 'just don't use envPrefix'. Hm.

Reply:
Is `ENV_PREFIX` available in `index.js`? Maybe we could do something like the followiing instead?
```js
if (ENV_PREFIX && process.env.LISTEN_PID !== undefined && env('LISTEN_PID') !== undefined) {
	throw new Error('...')
}
```

Reply:
Or actually, we could probably slightly modify `env.js`.

Reply:
Or actually, we could probably slightly modify `env.js`.
```js
const expected_unprefixed = new Set(['LISTEN_PID', 'LISTEN_FDS']);

export function env(name, fallback) {
	const prefix = expected_unprefixed.has(name) ? '' : ENV_PREFIX;
	const prefixed = prefix + name;
	return prefixed in process.env ? process.env[prefixed] : fallback;
}
```

Reply:
@Conduitry what do you think?

Reply:
Okay, as discussed on Discord I slightly modified `env.js` to ignore `envPrefix` for the systemd env vars. I added a note in the docs that `envPrefix` can be used even if socket activation is enabled.